modules/ipc/include/hephaestus/ipc/zenoh/action_server.h

```diff
@@ -0,0 +1,254 @@
+               ExecuteCallback&& execute_cb);
+  ~ActionServer();
+  ActionServer(const ActionServer&) = delete;
+  ActionServer(ActionServer&&) = delete;
+  auto operator=(const ActionServer&) -> ActionServer& = delete;
+  auto operator=(ActionServer&&) -> ActionServer& = delete;
+
+private:
+  [[nodiscard]] auto onRequest(const RequestT& request) -> ActionServerRequestResponse;
+  void execute(const RequestT& request);
+
+private:
+  SessionPtr session_;
+  TopicConfig topic_config_;
+
+  ActionTriggerCallback action_trigger_cb_;
+  ExecuteCallback execute_cb_;
+
+  std::unique_ptr<Service<RequestT, ActionServerRequestResponse>> request_service_;
+  concurrency::MessageQueueConsumer<RequestT> request_consumer_;
+  std::atomic_bool is_running_{ false };
+};
+
+template <typename StatusT>
+using StatusUpdateCallback = std::function<void(const StatusT&)>;
+
+/// Call the action server with the given request.
+/// - The status_update_cb will be called with the status updates from the server.
+///   - The updates are decided by the user implementation of the action server.
+/// - Returns a future which will eventually contain the response from the server.
+template <typename RequestT, typename StatusT, typename ReplyT>
+[[nodiscard]] auto callActionServer(SessionPtr session, const TopicConfig& topic_config,
+                                    const RequestT& request, StatusUpdateCallback<StatusT>&& status_update_cb)
+    -> std::future<ActionServerResponse<ReplyT>>;
+
+/// Request the action server to stop.
+[[nodiscard]] auto requestActionServerToStopExecution(Session& session,
+                                                      const TopicConfig& topic_config) -> bool;
+
+// TODO: add a function to get notified when the server is idle again.
+
+// ----------------------------------------------------------------------------------------------------------
+// --------- ActionServer Implementation --------------------------------------------------------------------
+// ----------------------------------------------------------------------------------------------------------
+
+template <typename RequestT, typename StatusT, typename ReplyT>
+ActionServer<RequestT, StatusT, ReplyT>::ActionServer(SessionPtr session, TopicConfig topic_config,
+                                                      ActionTriggerCallback&& action_trigger_cb,
+                                                      ExecuteCallback&& execute_cb)
+  : session_(std::move((session)))
+  , topic_config_(std::move(topic_config))
+  , action_trigger_cb_(std::move(action_trigger_cb))
+  , execute_cb_(std::move(execute_cb))
+  , request_service_(std::make_unique<Service<RequestT, ActionServerRequestResponse>>(
+        session_, topic_config_, [this](const RequestT& request) { return onRequest(request); }))
+  , request_consumer_([this](const RequestT& request) { return execute(request); }, std::nullopt) {
+  request_consumer_.start();
```

Comment:
nit. Given the terminology `execute_cb`, should this be called `executor` or similar?

I just read it as request something, i.e. request a consumer first.

Reply:
usualy function names are verb and class are actor, so executor would be the name if it were a class

Reply:
good point, leave it